ratis-server/src/main/java/org/apache/ratis/server/raftlog/segmented/SegmentedRaftLog.java

```diff
@@ -414,6 +414,11 @@ protected CompletableFuture<Long> appendEntryImpl(LogEntryProto entry) {
       } else {
         cache.appendEntry(entry, LogSegment.Op.WRITE_CACHE_WITHOUT_STATE_MACHINE_CACHE);
       }
+      writeFuture.whenComplete((clientReply, exception) -> {
+        if (appendEntryTimerContext != null) {
```

Comment:
After changed to `final`, it cannot be null.

Reply:
Done.